core/src/avm2/globals/flash/display/BitmapData.as

```diff
@@ -1,5 +1,56 @@
 // This is a stub - the actual class is defined in `bitmapdata.rs`
 package flash.display {
-    public class BitmapData {
+    import flash.geom.Rectangle;
+    import flash.geom.ColorTransform;
+    import flash.geom.Point;
+    import flash.geom.Matrix;
+    import flash.filters.BitmapFilter;
+    import flash.utils.ByteArray;
+
+    [Ruffle(InstanceAllocator)]
+    public class BitmapData implements IBitmapDrawable {
+        // FIXME - the first two arguments should not be defaulted, but it's currently
```

Comment:
Couldn't this be fixed by manually setting width and height to 0 based on arguments.length in the ActionScript part of the constructor?

Reply:
This PR is already quite large, so I wanted to do it separately.

Reply:
The real fix for this to create a native init method for `BitmapData`, and call that in `BitmapDataObject::from_bitmap_data`

Reply:
Site note: mentioned before on Discord, but I was wondering if maybe we should make `instance_allocator` be able to receive args (to be a bit more like C++ constructor), so we can avoid two-phase initialization in classes like BitmapData (and other DOs) or `dummy` stuff in the XML PR.